packages/x-date-pickers/src/internals/components/PickerFieldUI.tsx

```diff
@@ -0,0 +1,329 @@
+
+  const translations = usePickerTranslations();
+  const pickerContext = useNullablePickerContext();
+  const {
+    textFieldProps,
+    onClear,
+    clearable,
+    openPickerAriaLabel,
+    clearButtonPosition: clearButtonPositionProp = 'end',
+    openPickerButtonPosition: openPickerButtonPositionProp = 'end',
+  } = cleanFieldResponse(fieldResponse);
+  const ownerState = useFieldOwnerState(textFieldProps);
+
+  const handleTogglePicker = useEventCallback(() => pickerContext?.setOpen((prev) => !prev));
+
+  const triggerStatus = pickerContext ? pickerContext.triggerStatus : 'hidden';
+  const clearButtonPosition = clearable ? clearButtonPositionProp : null;
+  const openPickerButtonPosition = triggerStatus !== 'hidden' ? openPickerButtonPositionProp : null;
+
+  const TextField =
+    slots.textField ??
+    (fieldResponse.enableAccessibleFieldDOMStructure === false ? MuiTextField : PickersTextField);
+
+  const InputAdornment = slots.inputAdornment ?? MuiInputAdornment;
+  const { ownerState: startInputAdornmentOwnerState, ...startInputAdornmentProps } = useSlotProps({
+    elementType: InputAdornment,
+    externalSlotProps: slotProps?.inputAdornment,
+    additionalProps: {
+      position: 'start' as const,
+    },
+    ownerState: { ...ownerState, position: 'start' },
+  });
+  const { ownerState: endInputAdornmentOwnerState, ...endInputAdornmentProps } = useSlotProps({
+    elementType: InputAdornment,
+    externalSlotProps: slotProps?.inputAdornment,
+    additionalProps: {
+      position: 'end' as const,
+    },
+    ownerState: { ...ownerState, position: 'end' },
+  });
+
+  const OpenPickerButton = slots.openPickerButton ?? MuiIconButton;
+  // We don't want to forward the `ownerState` to the `<IconButton />` component, see mui/material-ui#34056
+  const {
+    ownerState: openPickerButtonOwnerState,
+    ...openPickerButtonProps
+  }: IconButtonProps & { ownerState: any } = useSlotProps({
+    elementType: OpenPickerButton,
+    externalSlotProps: slotProps?.openPickerButton,
+    additionalProps: {
+      disabled: triggerStatus === 'disabled',
+      onClick: handleTogglePicker,
+      'aria-label': openPickerAriaLabel,
+      edge:
+        clearButtonPosition === 'start' && openPickerButtonPosition === 'start'
+          ? undefined
+          : openPickerButtonPosition,
+    },
+    ownerState,
+  });
+
+  const OpenPickerIcon = slots.openPickerIcon;
+  const openPickerIconProps = useSlotProps({
+    elementType: OpenPickerIcon,
+    externalSlotProps: slotProps?.openPickerIcon,
+    ownerState,
+  });
+
+  const ClearButton = slots.clearButton ?? MuiIconButton;
+  // We don't want to forward the `ownerState` to the `<IconButton />` component, see mui/material-ui#34056
+  const { ownerState: clearButtonOwnerState, ...clearButtonProps } = useSlotProps({
+    elementType: ClearButton,
+    externalSlotProps: slotProps?.clearButton,
+    className: 'clearButton',
+    additionalProps: {
```

Comment:
You probably forgot to provide the `sx` prop? 🤔 
https://github.com/mui/mui-x/blob/3b0bba2453c8b7a052b2e434260a2f8d911f8c62/packages/x-date-pickers/src/hooks/useClearableField.tsx#L104-L119

In this branch, the clearable button is always visible, whereas currently, it is only shown on hover or when the input is focused.

The biggest question is how do we approach the clearable behavior on the Mobile Picker - what behavior would bring enough value to users on Mobile? 🤔